docs/source/stability_policy.rst

```diff
@@ -0,0 +1,132 @@
+Stability Policy
+================
+
+The ``python-telegram-bot`` library (PTB) tries to provides a stable interface to its users.
+As both the user base and the complexity of the library has grown significantly since PTBs conception, we introduced below explicit stability Policy with version NEXT.VERSION.
+
+Large parts of the :mod:`telegram` package are the Python representations of the Telegram Bot API, the stability policy of which PTB can not influence.
+This policy hence includes some special cases for those parts.
+
+What does this policy cover?
+----------------------------
+
+This policy includes any API or behavior that is documented in this documentation.
+This covers both the :mod:`telegram` package and the :mod:`telegram.ext` package.
+
+What doesn't this policy cover?
+-------------------------------
+
+Introduction of new features or changes of flavors of comparable behavior (e.g. the default for the used HTTP protocol version) are not covered by this policy.
+The internal structure of classes in PTB, i.e. things like the result of ``dir(obj))`` or the contents of ``obj.__dict__``, may change.
+
+Objects are in general not guaranteed to be pickleable (unless stated otherwise) and pickled objects from one version of PTB may not be loadable in future versions.
+We may provide a way to convert pickled objects from one version to another, but this is not guaranteed.
+
+Functionality that is part of PTBs API but is explicitly documented to not be intended to be used directly by users (e.g. :meth:`telegram.request.BaseRequest.do_request`) may change.
+This also applies to functions or attributes marked as final in the sense of `PEP 591 <https://www.python.org/dev/peps/pep-0591/>`__.
+
+PTB has dependencies to 3rd party packages.
+The versions that PTB uses of these 3rd party packages may change if that does not change PTBs public API.
+
+PTB does not give guarantees about which Python versions are supported.
+In general, we will try to support all Python versions that have not yet reached their end of life, but we reserve ourselves the option to drop support for Python versions earlier if that benefits the advancement of the library.
+
+.. _bot-api-functionality-1:
+
+Bot API Functionality
+~~~~~~~~~~~~~~~~~~~~~
+
+Comparison of equality of the classe in the :mod:`telegram` package is subject to change and the PTB team will update the behavior to best reflect updates in the Bot API.
+
+When arguments of the Bot API methods change order or become optional/mandatory due to changes in the Bot API, PTB will always try to reflect these changes.
```

Comment:
```suggestion
When the order of arguments of Bot API methods switch or they become optional/mandatory due to changes in the Bot API, PTB will always reflect these changes.
```
we dont try to reflect, we always reflect, no? Just in the least breaking way, see next paragraph

Reply:
You're referring to

> While we try to make such changes backwards compatible, this is not always possible or only with significant effort.
In such cases we will find a trade-off between backwards compatibility and fully complying with the Bot API, which may result in breaking changes.
We highly recommend using keyword arguments, which can help to make such changes non-breaking on your end.

I guess? 

IMO we don't always fully reflect the Bot API. (That ofc also depends a bit on what one means by that). Best example is the order of the optional arguments in PTB differs from the Bot API quite often. Also we recently didn't remove the `thumb` arguments immediately which can also be seen as "not (fully) reflecting the changes".

Stating here that we always fully reflect the changes in the Bot API is IMO a contradiction to the paragraph that I quoted above …

Reply:
applied the wording changes for now

Reply:
I like it like that